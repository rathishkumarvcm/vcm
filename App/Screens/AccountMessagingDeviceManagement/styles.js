import { StyleSheet } from 'react-native';
import { scaledHeight } from '../../Utils/Resolution';

const styles = StyleSheet.create({
    cancelButton: {
        alignItems: 'center',
        backgroundColor: '#FFFFFF',
        borderColor: '#7B8288',
        borderWidth: 0.5,
        height: scaledHeight(55),
        justifyContent: 'center',
        marginLeft: '10%',
        marginRight: '10%',
        marginTop: scaledHeight(28),
        width: '80%',
    },
    cancelButtonText: {
        color: '#544A54',
        fontSize: scaledHeight(16),
        fontWeight: 'bold'
    },
    container: {
        backgroundColor: '#F7FAFF',
        flex: 1,
        width: '100%',
    },
    deviceAlertsContent: {
        color: '#707070',
        fontSize: scaledHeight(14),
        marginLeft: '6%',
        marginTop: scaledHeight(18),
    },
    deviceHeaderViewTitle: {
        color: '#707070',
        fontSize: scaledHeight(18),
        marginLeft: '3%',
    },
    deviceHeaderview: {
        alignItems: 'center',
        backgroundColor: '#F1F1F2',
        flexDirection: 'row',
        paddingBottom: scaledHeight(14),
        paddingLeft: '2%',
        paddingTop: scaledHeight(14),
        width: '100%',
    },
    devicePushNotifyText: {
        color: '#707070',
        fontSize: scaledHeight(14),
        marginBottom: scaledHeight(8),
        marginLeft: '4%',
        marginRight: '4%',
        marginTop: scaledHeight(14)
    },  
    deviceSectionGrp: {
        flexGrow: 1,
        marginLeft: '4%',        
    },
    deviceSectionTxt: {
        color: '#333333DE',
        fontSize: scaledHeight(14),
        fontWeight: 'bold',
        marginLeft: '4%',       
        textAlign: 'left',     
    },
    devicecontainer: {
        backgroundColor: '#FFFFFF',
        borderColor: '#EBEDEF',
        borderWidth: 0.5,
        marginLeft: '4%',
        marginRight: '4%',
        marginTop: scaledHeight(24),
        width: '92%',
    },
    devicecontainerBottom: {
        backgroundColor: '#FFFFFF',
        borderColor: '#EBEDEF',
        borderTopWidth: 0.5,
        borderWidth: 1,
        marginLeft: '4%',
        marginRight: '4%',
        width: '92%',
    },
    lastUsedText: {       
        color: '#707070',
        fontSize: scaledHeight(14),      
        marginBottom: scaledHeight(8),
        marginLeft: '6%',      
    },
    saveButton: {
        alignItems: 'center',
        backgroundColor: '#56565A',
        borderColor: '#56565A',
        borderWidth: 0.5,
        height: scaledHeight(55),
        justifyContent: 'center',
        marginLeft: '10%',
        marginRight: '10%',
        marginTop: scaledHeight(25),
        width: '80%',
    },
    saveButtonText: {
        color: '#FFFFFF',
        fontSize: scaledHeight(16),
        fontWeight: 'bold'
    },
    scrollViewFlex:{
        flex: 0.85
    },
    settingsInfo: {
        color: '#B2B2B2',
        fontSize: scaledHeight(13),
        marginRight: '4%',
    },  
    settingsInfoCurrent: {
        color: '#707070',
        fontSize: scaledHeight(14),
        fontWeight: 'bold',
    },
    settingsInfoHead: {
        marginTop: scaledHeight(18),
        paddingLeft: '4%',
        paddingRight: '4%',
        width: '100%',
    },
    settingsInfoHeadTilte: {
        color: '#56565A',
        fontSize: scaledHeight(20),
        fontWeight: 'bold',
        paddingBottom: scaledHeight(4)
    },
    settingsView: {
        flexDirection: 'row',
        flexWrap: 'wrap',
        marginTop: scaledHeight(18),
        paddingLeft: '4%',
        paddingRight: '4%',
        width: '100%',
    },
    touchOpacityPosition: {
        position: 'relative',
    },

});
export default styles;