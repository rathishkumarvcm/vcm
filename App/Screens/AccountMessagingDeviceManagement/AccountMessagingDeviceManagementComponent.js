import React, { Component } from 'react';
import { Text, View, ScrollView, TouchableOpacity } from 'react-native';
import { styles } from './styles';
import { GHeaderComponent, GFooterSettingsComponent, GIcon, GButtonComponent } from '../../CommonComponents';
import PropTypes from 'prop-types';
import { CustomCheckBox } from '../../AppComponents';
import gblStrings from '../../Constants/GlobalStrings';

class AccountMessagingDeviceManagementComponent extends Component {
    constructor(props) {
        super(props);
        //set true to isLoading if data for this screen yet to be received and wanted to show loader.
        this.state = {
            isLoading: false,
            deviceSelected: false
        };
    }

    goBack = () => {
        this.props.navigation.goBack();
    }

    saveButtonAction = () => {        
        console.log('Save Button Clicked...');       
        const payloadData = {
            deviceSelected: this.state.deviceSelected
        };
        this.props.saveData(payloadData);
        this.props.navigation.goBack();                
        
    }

    setDeviceState = () => {     
        this.setState({deviceSelected : !this.state.deviceSelected });
    }

    navigategeneralSettings = () => this.props.navigation.navigate('generalSettings');

    componentDidMount() {
        if (this.props && this.props.accMessageDeviceinitialState){
            (this.props.accMessageDeviceinitialState.deviceSelected);
                this.setState({ deviceSelected : this.props.accMessageDeviceinitialState.deviceSelected });
        }        
    }
   
    render() {
        return (
            <View style={styles.container}>
                <GHeaderComponent
                    navigation={this.props.navigation}
                />
                <ScrollView style={{ flex: 0.85 }}>
                    <View style={styles.settingsView}>
                        <TouchableOpacity style={styles.touchOpacityPosition} onPress={this.navigategeneralSettings}>
                            <Text style={styles.settingsInfo}>
                                {gblStrings.settingAccountMessaging.setting}
                            </Text>
                        </TouchableOpacity>
                        <TouchableOpacity style={styles.touchOpacityPosition} onPress={this.goBack}>
                            <Text style={styles.settingsInfo}>
                                {gblStrings.settingAccountMessaging.accountMessagingArrow}
                            </Text>
                        </TouchableOpacity>
                        <Text style={styles.settingsInfoCurrent}>
                            {gblStrings.settingAccountMessaging.accountMessagingDeviceManageArrow}
                        </Text>
                    </View>

                    <View style={styles.settingsInfoHead}>
                        <Text style={styles.settingsInfoHeadTilte}>
                            {gblStrings.settingAccountMessaging.deviceManagementTitle}
                        </Text>
                        <Text style={styles.settingsInfo}>
                            {gblStrings.settingAccountMessaging.accountMessagingDeviceManageTitleDesc}
                        </Text>
                    </View>

                    <View style={styles.devicecontainer}>
                        <View style={styles.deviceHeaderview}>
                            <GIcon
                                name="minus"
                                type="antdesign"
                                size={30}
                                color="#088ACC"
                            />
                            <Text style={styles.deviceHeaderViewTitle}>
                                {gblStrings.settingAccountMessaging.deviceManagementTitle}
                            </Text>
                        </View>
                    </View>
                    <View style={styles.devicecontainerBottom}>
                        <Text style={styles.devicePushNotifyText}>
                            {gblStrings.settingAccountMessaging.accountMessagingDeviceManagePushNotify}
                        </Text>
                        <Text style={styles.devicePushNotifyText}>
                            {gblStrings.settingAccountMessaging.accountMessagingDeviceManageDeviceRemove}
                        </Text>

                        <View style={styles.deviceSectionGrp} >
                            <CustomCheckBox
                                size={20}
                                itemBottom={0}
                                itemTop={0}
                                outerCicleColor={"#707070"}
                                innerCicleColor={"#61285F"}
                                labelStyle={styles.deviceSectionTxt}
                                label={'iPhone XR'}
                                selected={this.state.deviceSelected}
                                onPress={this.setDeviceState}
                            />                            
                        </View>                              
                        <Text style={styles.lastUsedText}>{'        last used : 08/08/2019'}</Text>   
                    </View>
                   
                    <GButtonComponent
                        buttonStyle={styles.cancelButton}
                        buttonText={gblStrings.common.cancel}
                        textStyle={styles.cancelButtonText}
                        onPress={this.goBack}
                    />
                    <GButtonComponent
                        buttonStyle={styles.saveButton}
                        buttonText={gblStrings.common.save}
                        textStyle={styles.saveButtonText}
                        onPress={this.saveButtonAction}
                    />
                    <GFooterSettingsComponent />
                </ScrollView>
            </View>
        );
    }
}

AccountMessagingDeviceManagementComponent.propTypes = {
    navigation: PropTypes.instanceOf(Object)
};

AccountMessagingDeviceManagementComponent.defaultProps = {

};

export default AccountMessagingDeviceManagementComponent;
