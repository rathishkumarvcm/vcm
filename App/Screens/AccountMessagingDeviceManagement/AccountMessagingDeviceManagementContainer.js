import { connect } from 'react-redux';
import AccountMessagingDeviceManagementComponent from './AccountMessagingDeviceManagementComponent';
import { getLogin } from '../../Reducers';
import { loginActions } from '../../Actions';

/*----------------------
                                  Redux Methods
                                                             -------------------------- */

const mapStateToProps = (state /* , props */) => ({
  dashboardData: getLogin(state),
});

const mapDispatchToProps = {
  ...loginActions,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AccountMessagingDeviceManagementComponent);
