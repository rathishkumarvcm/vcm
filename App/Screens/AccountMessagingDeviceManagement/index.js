import AccountMessagingDeviceManagementComponent from './AccountMessagingDeviceManagementContainer';
export default AccountMessagingDeviceManagementComponent;
