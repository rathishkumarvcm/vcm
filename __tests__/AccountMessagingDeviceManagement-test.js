import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import AccountMessagingDeviceManagement from '../App/Screens/AccountMessagingDeviceManagement/AccountMessagingDeviceManagementComponent';

describe('Account Messaging Device Management',()=>{

    // Snap shot 
    it('Match Snapshot', () => {  
      const tree = renderer.create(<AccountMessagingDeviceManagement />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    // Render using shallow - shallow not includes child component
    test('render',()=>{
      const component = shallow(<AccountMessagingDeviceManagement />);  
      component.render();
    });

});