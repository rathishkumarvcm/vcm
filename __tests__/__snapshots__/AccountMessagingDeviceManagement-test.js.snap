// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Messaging Device Management Match Snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F7FAFF",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    clickable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 0.12,
        "flexDirection": "row",
        "opacity": 1,
        "paddingLeft": "4%",
        "paddingRight": "4%",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100.5,
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../App/Images/logo.png",
          }
        }
        style={
          Object {
            "alignItems": "flex-start",
            "justifyContent": "flex-start",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": "12%",
        }
      }
    />
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 24.5,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "width": 56,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 40,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "EvilIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "10%",
        }
      }
    />
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "width": 44,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "width": 44,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "alignItems": "flex-end",
              "justifyContent": "center",
              "marginEnd": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 40,
                "width": 56,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 40,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "marginHorizontal": 5,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 0.85,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "marginTop": 29.5,
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "relative",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#B2B2B2",
                "fontSize": 21.5,
                "marginRight": "4%",
              }
            }
          >
            Setting
          </Text>
        </View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "relative",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#B2B2B2",
                "fontSize": 21.5,
                "marginRight": "4%",
              }
            }
          >
            &gt;    Account Messaging
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#707070",
              "fontSize": 23,
              "fontWeight": "bold",
            }
          }
        >
          &gt;     Device Management
        </Text>
      </View>
      <View
        style={
          Object {
            "marginTop": 29.5,
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 33,
              "fontWeight": "bold",
              "paddingBottom": 6.5,
            }
          }
        >
          Device Management
        </Text>
        <Text
          style={
            Object {
              "color": "#B2B2B2",
              "fontSize": 21.5,
              "marginRight": "4%",
            }
          }
        >
          Lorem Ipsum is dummy text of the typesetting industry.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#EBEDEF",
            "borderWidth": 0.5,
            "marginLeft": "4%",
            "marginRight": "4%",
            "marginTop": 39.5,
            "width": "92%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F1F1F2",
              "flexDirection": "row",
              "paddingBottom": 23,
              "paddingLeft": "2%",
              "paddingTop": 23,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 44,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#088ACC",
                    "fontSize": 30,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "marginHorizontal": 5,
                  },
                  Object {
                    "fontFamily": "anticon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#707070",
                "fontSize": 29.5,
                "marginLeft": "3%",
              }
            }
          >
            Device Management
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#EBEDEF",
            "borderTopWidth": 0.5,
            "borderWidth": 1,
            "marginLeft": "4%",
            "marginRight": "4%",
            "width": "92%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#707070",
              "fontSize": 23,
              "marginBottom": 13,
              "marginLeft": "4%",
              "marginRight": "4%",
              "marginTop": 23,
            }
          }
        >
          Push notifications are automatically set up on the device you are using every time you access the VCM mobile app.
        </Text>
        <Text
          style={
            Object {
              "color": "#707070",
              "fontSize": 23,
              "marginBottom": 13,
              "marginLeft": "4%",
              "marginRight": "4%",
              "marginTop": 23,
            }
          }
        >
          Select the Devices you wish to remove.
        </Text>
        <View
          style={
            Object {
              "flexGrow": 1,
              "marginLeft": "4%",
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexGrow": 1,
                "justifyContent": "flex-start",
                "marginBottom": 0,
                "marginTop": 0,
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#707070",
                  "borderWidth": 2,
                  "height": 20,
                  "justifyContent": "center",
                  "marginTop": 3.5,
                  "width": 20,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "marginLeft": 19.5,
                    "width": "80%",
                  },
                  Object {
                    "color": "#333333DE",
                    "fontSize": 23,
                    "fontWeight": "bold",
                    "marginLeft": "4%",
                    "textAlign": "left",
                  },
                ]
              }
            >
              iPhone XR
            </Text>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#707070",
              "fontSize": 23,
              "marginBottom": 13,
              "marginLeft": "6%",
            }
          }
        >
                  last used : 08/08/2019
        </Text>
      </View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#7B8288",
            "borderRadius": 1,
            "borderWidth": 0.5,
            "height": 90.5,
            "justifyContent": "center",
            "marginLeft": "10%",
            "marginRight": "10%",
            "marginTop": 46,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#544A54",
                "fontSize": 26.5,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Cancel
        </Text>
      </View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#56565A",
            "borderColor": "#56565A",
            "borderRadius": 1,
            "borderWidth": 0.5,
            "height": 90.5,
            "justifyContent": "center",
            "marginLeft": "10%",
            "marginRight": "10%",
            "marginTop": 41,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 26.5,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Save
        </Text>
      </View>
      <View>
        <View
          style={
            Object {
              "borderColor": "#DEDEDF",
              "borderTopWidth": 1,
              "marginTop": 65.5,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
            }
          }
        >
          <View
            style={
              Object {
                "marginBottom": 41,
                "marginTop": 41,
                "paddingLeft": "4%",
                "paddingRight": "4%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#56565A",
                  "fontSize": 29.5,
                  "fontWeight": "bold",
                }
              }
            >
              Investing involves risk including loss of principal.
            </Text>
            <Text
              style={
                Object {
                  "color": "#56565A",
                  "fontSize": 26.5,
                  "lineHeight": 33,
                  "marginTop": 24.5,
                }
              }
            >
              Victory Mutual Funds and USAA Mutual Funds are distributed by Victory Capital Advisers, Inc. (VCA). VictoryShares ETFs and VictoryShares USAA ETFs are distributed by Foreside Fund Services, LLC (Foreside). VCA and Foreside are members of FINRA and SIPC. Victory Capital Management Inc. (VCM) is the investment adviser to the Victory Mutual Funds, USAA Mutual Funds, VictoryShares ETFs and VictoryShares USAA ETFs. VCA and VCM are not affiliated with Foreside. USAA is not affiliated with Foreside, VCM, or VCA. USAA and the USAA logos are registered trademarks and the USAA Mutual Funds and USAA Investments logos are trademarks of United Services Automobile Association and are being used by Victory Capital and its affiliates under license. Victory Capital means Victory Capital Management Inc., the investment manager of the USAA 529 College Savings Plan (Plan). The Plan is distributed by Victory Capital Advisers, Inc., a broker dealer registered with FINRA and an affiliate of Victory Capital. Victory Capital and its affiliates are not affiliated with United Services Automobile Association or its affiliates. USAA and the USAA logo are registered trademarks and the USAA 529 College Savings Plan logo is a trademark of United Services Automobile Association and are being used by Victory Capital and its affiliates under license.
            </Text>
          </View>
          <View
            style={
              Object {
                "marginBottom": 41,
                "marginTop": 33,
                "paddingLeft": "4%",
                "paddingRight": "4%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../App/Images/logo.png",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "marginLeft": "4%",
                "marginTop": 33,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#56565A",
                  "fontSize": 26.5,
                  "lineHeight": 33,
                  "marginTop": 24.5,
                }
              }
            >
              Connect with Us
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 33,
                "marginLeft": "2%",
                "marginTop": 16.5,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 35,
                  "width": 50,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#56565A",
                      "fontSize": 35,
                    },
                    Object {
                      "backgroundColor": "transparent",
                      "marginHorizontal": 5,
                    },
                    Object {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": 35,
                  "width": 50,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#56565A",
                      "fontSize": 35,
                    },
                    Object {
                      "backgroundColor": "transparent",
                      "marginHorizontal": 5,
                    },
                    Object {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "height": 65.5,
                "justifyContent": "center",
                "marginLeft": "2%",
                "marginTop": 24.5,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#5D83AE",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "paddingLeft": "2%",
                  "width": "50%",
                }
              }
            >
              Privacy Policy
            </Text>
            <Text
              style={
                Object {
                  "color": "#5D83AE",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "paddingLeft": "2%",
                  "width": "50%",
                }
              }
            >
              Fund Documents
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "height": 65.5,
                "justifyContent": "center",
                "marginLeft": "2%",
                "marginTop": 24.5,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#5D83AE",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "paddingLeft": "2%",
                  "width": "50%",
                }
              }
            >
              User Agreements
            </Text>
            <Text
              style={
                Object {
                  "color": "#5D83AE",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "paddingLeft": "2%",
                  "width": "50%",
                }
              }
            >
              Support
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#56565A",
                "height": 98.5,
                "justifyContent": "center",
                "marginTop": 33,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                }
              }
            >
              Copyright Victory Capital Management Inc. ©2020
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
